data_steward/validation/participants/identity_match.py

```diff
@@ -618,70 +620,78 @@ def _get_date_string(dataset):
     return datetime.now().strftime(consts.DRC_DATE_FORMAT)
 
 
-def write_results_to_site_buckets(project, validation_dataset=None):
+def write_results_to_site_buckets(storage_client, validation_dataset):
     """
     Write the results of participant matching to each site's bucket.
 
-    :param project: a string representing the project name
+    :param storage_client: a client to generate a bucket for a blob
     :param validation_dataset:  the identifier for the match values
         destination dataset
 
     :return: None
     :raises:  RuntimeError if validation_dataset is not defined.
     """
     LOGGER.info('Writing to site buckets')
-    if validation_dataset is None:
-        LOGGER.error('Validation_dataset name is not defined.')
-        raise RuntimeError('validation_dataset name cannot be None.')
+    if not validation_dataset:
+        LOGGER.error('Validation dataset name is not defined.')
+        raise RuntimeError(f"Validation_dataset is '{validation_dataset}'")
 
-    date_string = _get_date_string(validation_dataset)
-    hpo_sites = readers.get_hpo_site_names()
+    date: str = _get_date_string(validation_dataset)
+    hpo_sites: list = readers.get_hpo_site_names()
     # generate hpo site reports
     for site in hpo_sites:
-        bucket = gcs_utils.get_hpo_bucket(site)
-        filename = os.path.join(
-            consts.REPORT_DIRECTORY.format(date=date_string),
-            consts.REPORT_TITLE)
-        _, errors = writers.create_site_validation_report(
-            project, validation_dataset, [site], bucket, filename)
-
-        if errors > 0:
+        filename: str = os.path.join(consts.REPORT_DIRECTORY.format(date=date),
+                                     consts.REPORT_TITLE)
+        try:
+            bucket = storage_client.get_hpo_bucket(site)
+            blob = bucket.blob(filename)
+        except GoogleCloudError as exc:
+            LOGGER.exception(
+                f'Encountered {exc.message} error when uploading site report')
```

Comment:
Nice, this ensures other sites can continue to be written to if one fails

Reply:
I had to double-check that we can use `exc.message` instead of `exc`, so this won't raise while raising --  `exc.message` won't raise an attribute error.

Reply:
Oh right, this is something that I discussed with @lledezma as well for BucketNotSet. Adding a message attribute helped with custom exceptions. Google Cloud seems to have it already set for their exceptions.